walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,97 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (request.getDuration() > 1440)
+      return Collections.emptyList();
+
+    // TreeSet has O(log n) insertion time
+    SortedSet<TimeRange> mandatoryBlockers = new TreeSet<>(TimeRange.ORDER_BY_START);
+
+    // which makes this loop an O(n log n) operation
+    events: for (Event event : events) {
+      for (String attendee : event.getAttendees()) {
+        if (request.getAttendees().contains(attendee)) {
+          // this is a blocker, split up the available time segment
+          mandatoryBlockers.add(event.getWhen());
+          continue events;
+        }
+      }
+    }
+
+    // mergeTimeRanges is O(n)
```

Comment:
This and the next comment are unnecessary. The code is self explanatory given the function names and variable names. We don't need to list time complexity of every operation, just the important ones where you made deliberate and potentially non-intuitive decisions. The goal of a comment is often to keep future developers from making mistakes that have nuanced and unnoticed consequences.